src/hotspot/share/code/compressedStream.cpp

```diff
@@ -70,26 +79,30 @@ CompressedWriteStream::CompressedWriteStream(int initial_size) : CompressedStrea
 }
 
 void CompressedWriteStream::grow() {
-  u_char* _new_buffer = NEW_RESOURCE_ARRAY(u_char, _size * 2);
+  int nsize = _size * 2;
+  const int min_expansion = UNSIGNED5::MAX_LENGTH;
+  if (nsize < min_expansion*2)
+    nsize = min_expansion*2;
```

Comment:
It's not clear if this is needed or just an optimization.  Maybe add a comment.  Also, using MIN2 might be clearer.